### ISSUE TITLE
IllegalArgumentException when all embedding fields not shown or doing a partial update without embedding fields

### ISSUE THREAD

**Issue by zane-neo** [opensearch-project/neural-search#73]
### What is the bug?
When all embedding fields not shown in the document or doing a partial update without embedding fields, an IllegalArgumentException will raise as below:
```
{
    "error": {
        "root_cause": [
            {
                "type": "illegal_argument_exception",
                "reason": "empty docs"
            }
        ],
        "type": "illegal_argument_exception",
        "reason": "empty docs"
    },
    "status": 400
}
```

### How can one reproduce the bug?
This issue is reported here: https://forum.opensearch.org/t/feedback-neural-search-plugin-experimental-release/11501/8.
First to create an ingest pipeline and an index as below:
```
PUT _ingest/pipeline/nlp-pipeline
{
    "description": "An example neural search pipeline",
    "processors": [
        {
            "text_embedding": {
                "model_id": "kI6NhoQB3oLQzIJTkldg",
                "field_map": {
                    "text": "text_knn"
                }
            }
        }
    ]
}

PUT /my-nlp-index-1
{
    "settings": {
        "index.knn": true,
        "default_pipeline": "nlp-pipeline"
    },
    "mappings": {
        "properties": {
            "passage_embedding": {
                "type": "knn_vector",
                "dimension": 384,
                "method": {
                    "name": "hnsw",
                    "space_type": "l2",
                    "engine": "nmslib",
                    "parameters": {
                        "ef_construction": 128,
                        "m": 24
                    }
                }
            },
            "passage_text": {
                "type": "text"
            }
        }
    }
}
```
* When inserting a new document
```
PUT my-nlp-index-1/_doc/1
{
    "name": "doc1"
}
```
* When updating a new document
First to ingest a document successfully:
```
PUT my-nlp-index-1/_doc/1
{
    "text": "doc1"
}
```
Then do a partial update to this document:
```
PUT my-nlp-index-1/_doc/1
{
    "name": "doc2"
}
```

### What is the expected behavior?
No exception raise and document should be ingested successfully.

### What is your host/environment?
Irrelevant with environment.

**Comment by navneet1v** [comment]
@zane-neo can we work on the fix and may be launch this in 2.4.1 as a patch release?

**Comment by navneet1v** [comment]
Closing the GH issue, as the changes are merged and release as part of 2.4.1 patch
